pkg/tpm/structures.go

```diff
@@ -126,32 +126,60 @@ type TcgBiosSpecIDEvent struct {
 
 // TcgPcrEvent2 is a TPM2 default log structure (EFI only)
 type TcgPcrEvent2 struct {
-	pcrIndex  uint32 `json:"index"`
+	pcrIndex  uint32
 	eventType uint32
 	digests   LDigestValues
 	eventSize uint32
 	event     []byte
 }
 
-func (e *TcgPcrEvent2) MarshalJSON() ([]byte, error) {
+type MDigest struct {
+	algorithm string
+	digest    string
+}
+
+func (md *MDigest) MarshalJSON() ([]byte, error) {
 	m := make(map[string]string)
-	m["type"] = e.PcrEventName()
+	m["algorithm"] = md.algorithm
+	m["digest"] = md.digest
+	return json.Marshal(m)
+}
+
+// This could be a ton nicer, probably
+func (e *TcgPcrEvent2) MarshalJSON() ([]byte, error) {
+	m := make(map[string]json.RawMessage)
+	j, err := json.Marshal(e.PcrEventName())
+	if err != nil {
+		return nil, err
+	}
+	m["type"] = json.RawMessage(j)
 	d := e.PcrEventData()
 	if d != "" {
-		m["data"] = d
+		j, err := json.Marshal(d)
+		if err != nil {
+			return nil, err
+		}
+		m["data"] = json.RawMessage(j)
+	}
+	var ds []MDigest
+	for i := uint32(0); i < e.digests.count; i++ {
```

Comment:
nit: use `for _, d := range e.digests.digests { }`

Reply:
Thank you!